packages/google_maps_flutter/google_maps_flutter_android/android/src/test/java/io/flutter/plugins/googlemaps/ConvertTest.java

```diff
@@ -665,33 +660,124 @@ public void ConvertInterpretHeatmapOptionsReturnsCorrectData() {
+    LatLngBounds bounds = new LatLngBounds(southwest, northeast);
+    GroundOverlay groundOverlay = mock(GroundOverlay.class);
+    when(groundOverlay.getPosition()).thenReturn(position);
+    when(groundOverlay.getBounds()).thenReturn(bounds);
+
+    Messages.PlatformDoublePair anchor = Convert.buildGroundOverlayAnchorForPigeon(groundOverlay);
+
+    Assert.assertEquals(0.5, anchor.getX(), 1e-15);
+    Assert.assertEquals(0.5, anchor.getY(), 1e-15);
   }
 
-  // Helper method to generate 1x1 pixel base64 encoded png test image
-  private String generateBase64Image() {
-    int width = 1;
-    int height = 1;
-    Bitmap bitmap = Bitmap.createBitmap(width, height, Bitmap.Config.ARGB_8888);
-    Canvas canvas = new Canvas(bitmap);
+  @Test
+  public void buildGroundOverlayAnchorForPigeonWithCrossingMeridian() {
+    LatLng position = new LatLng(10, -175);
+    LatLng southwest = new LatLng(5, 170);
+    LatLng northeast = new LatLng(15, -160);
+    LatLngBounds bounds = new LatLngBounds(southwest, northeast);
+    GroundOverlay groundOverlay = mock(GroundOverlay.class);
+    when(groundOverlay.getPosition()).thenReturn(position);
+    when(groundOverlay.getBounds()).thenReturn(bounds);
 
-    // Draw on the Bitmap
-    Paint paint = new Paint();
-    paint.setColor(Color.parseColor("#FF8080FF"));
-    canvas.drawRect(0, 0, width, height, paint);
+    Messages.PlatformDoublePair anchor = Convert.buildGroundOverlayAnchorForPigeon(groundOverlay);
 
-    // Convert the Bitmap to PNG format
-    ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
-    bitmap.compress(Bitmap.CompressFormat.PNG, 100, outputStream);
-    byte[] pngBytes = outputStream.toByteArray();
+    Assert.assertEquals(0.5, anchor.getX(), 1e-15);
+    Assert.assertEquals(0.5, anchor.getY(), 1e-15);
+  }
 
-    // Encode the PNG bytes as a base64 string
-    return Base64.encodeToString(pngBytes, Base64.DEFAULT);
+  @Test
+  public void groundOverlayToPigeonWithPosition() {
+    GroundOverlay mockGroundOverlay = mock(GroundOverlay.class);
+    LatLng position = new LatLng(10, 20);
+    LatLng southwest = new LatLng(5, 15);
+    LatLng northeast = new LatLng(15, 25);
+    LatLngBounds bounds = new LatLngBounds(southwest, northeast);
+    when(mockGroundOverlay.getPosition()).thenReturn(position);
+    when(mockGroundOverlay.getBounds()).thenReturn(bounds);
+    when(mockGroundOverlay.getWidth()).thenReturn(30f);
+    when(mockGroundOverlay.getHeight()).thenReturn(40f);
+    when(mockGroundOverlay.getBearing()).thenReturn(50f);
+    when(mockGroundOverlay.getTransparency()).thenReturn(0.6f);
+    when(mockGroundOverlay.getZIndex()).thenReturn(7f);
+    when(mockGroundOverlay.isVisible()).thenReturn(true);
+    when(mockGroundOverlay.isClickable()).thenReturn(false);
+
+    Messages.PlatformGroundOverlay result =
+        Convert.groundOverlayToPigeon(mockGroundOverlay, "overlay_1", false);
+
+    Assert.assertEquals("overlay_1", result.getGroundOverlayId());
```

Comment:
Consider extracting a helper method that takes a actual ground overlay object and expected values and handles the assertions for equals. It would make new tests easier to write/review.